packages/mobile/src/components/payment-method/PaymentMethod.tsx

```diff
@@ -72,12 +78,22 @@ export const PaymentMethod = ({
 }: PaymentMethodProps) => {
   const styles = useStyles()
   const neutral = useColor('neutral')
+  const dispatch = useDispatch()
 
   const balanceCents = formatUSDCWeiToFloorCentsNumber(
     (balance ?? new BN(0)) as BNUSDC
   )
   const balanceFormatted = formatCurrencyBalance(balanceCents / 100)
   const purchaseVendor = useSelector(getPurchaseVendor)
+  const { isEnabled: isCoinflowEnabled, isLoaded: isCoinflowEnabledLoaded } =
+    useFeatureFlag(FeatureFlags.BUY_WITH_COINFLOW)
+
+  // Initial state is coinflow by default, but set to stripe if coinflow is disabled.
+  useEffect(() => {
```

Comment:
Since redux is the source of truth and can technically be interacted with from multiple UI components, it might make sense to check the flag and change the default value there? Worst case, we'd need an `initialize` action so a saga could grab context and read the flag before setting the values.

Reply:
spoke offline - the problem with current approach is that the component might flash the wrong value on load. To do it "right" it would need to be a saga. Going to merge this for now though.